File: Programmers/Inho/Level01/두 개 뽑아서 더하기.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public int[] solution(int[] numbers)
    {
        int[] answer = new int[] { };
        List<int> plus = new List<int>();
        for (int i = 0; i < numbers.Length; i++)
        {
            for (int j = i + 1; j < numbers.Length; j++)
            {
                plus.Add(numbers[i] + numbers[j]);
            }
        }
        plus = plus.Distinct().ToList();
        plus.Sort();
        answer = plus.ToArray();

        return answer;
    }
}

File: Programmers/Inho/C++/Level01/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    vector<int>::iterator it1;
    vector<int>::iterator it2;
    for(it1 = numbers.begin();it1!=numbers.end();it1++){
        for(it2 = it1+1;it2!=numbers.end();it2++){
            answer.push_back(*it1+*it2);
        }
    }
    sort(answer.begin(), answer.end());
    answer.erase(unique(answer.begin(), answer.end()), answer.end());
    return answer;
}